android/app/src/main/java/org/uwaterloo/subletr/pages/listingdetails/ListingDetailsPageViewModel.kt

```diff
@@ -102,4 +115,20 @@ class ListingDetailsPageViewModel @Inject constructor(
 		}
 		return@combineLatest ListingDetailsPageUiState.Loading
 	}
+
+	private fun favouriteListing(favouriteListingRequest: FavouriteListingRequest) {
+		viewModelScope.launch {
```

Comment:
I'm getting fond of `viewModelScope.launch`
since it wasn't in the code before was there a reason why we didn't use it?

Reply:
First, you need to use `launch(Dispatchers.Default)` to ensure that this is done `async`.
Just calling `launch` will block the main thread.

With respect to why we didn't use it before, it's mainly because we really don't need to...
Generally, `rxJava` gives us everything that we need and there is no reason to use `viewModelScope.launch` since `rxJava` already disposes stuff automatically as long you subscribe properly.